parlai/mturk/tasks/acute_eval/analysis.py

```diff
@@ -412,26 +481,71 @@ def _render_row(matchup: List[str], row: pd.Series) -> str:
+                '<label for= "neutral">Neutral</label>'
+                '</div>'
+                '</td>'
+            )
+            dialogue_row = f"<td>{dialogues['winner_dialogue']}</td><td>{dialogues['loser_dialogue']}</td>"
+            reason_row = f"<td>{row['reason']}\n{speakers_footnote}</td>"
+            if cherry_pick:
+                return f"<tr><td>Pair {str(row_id)}</td>{checkbox_row}{dialogue_row}{reason_row}</tr>"
+            else:
+                return f"<tr><td>Pair {str(row_id)}</td>{dialogue_row}{reason_row}</tr>"
+
+        def _render_html(table: pd.DataFrame, cherry_pick: bool) -> HTML:
+            result = '\
+                <div id="toc_container">\
+                <p class="toc_title">Model Pairs</p>\
+                <ul class="toc_list">'
+            for matchup in matchups:
+                eval_question = table.loc[table['matchup'] == matchup, 'question'].iloc[
+                    0
+                ]
+                result += f"<li><a href='#{matchup}''>{matchup + '__on__' +eval_question}</a></li>"
+            result += '</ul></div>'
             for matchup in matchups:
-                length = min(10, len(table[table['matchup'] == matchup]))
+                length = min(
+                    self.max_matchups_html, len(table[table['matchup'] == matchup])
+                )
+                eval_question = table.loc[table['matchup'] == matchup, 'question'].iloc[
+                    0
+                ]
                 matchup_table = table[table['matchup'] == matchup][:length]
                 table_rows = [
-                    _render_row(matchup.split('__vs__'), row)
-                    for i, row in matchup_table.iterrows()
+                    _render_row(matchup.split('__vs__'), row, idx, cherry_pick)
+                    for idx, (_, row) in enumerate(matchup_table.iterrows())
                 ]
-                table_body = f"<table><tr><th>Winner Conversation</th><th>Loser Conversation</th><th>Reason</th></tr>{''.join(table_rows)}</table>"
-                result += f"<h2>{matchup}</h2><body>{table_body}</body>"
+                if cherry_pick:
+                    table_body = f"<table border=1 frame=void rules=rows cellpadding='20'><tr><th>Pair Id</th><th>Comments</th><th>Winner Conversation</th><th>Loser Conversation</th><th>Reason</th></tr>{''.join(table_rows)}</table>"
+                else:
+                    table_body = f"<table border=1 frame=void rules=rows cellpadding='20'><tr><th>Pair Id</th><th>Winner Conversation</th><th>Loser Conversation</th><th>Reason</th></tr>{''.join(table_rows)}</table>"
+                result += f"<h2 id='{matchup}'><li><a href='#toc_container'>{matchup + '__on__' +eval_question}</a></li></h2><body>{table_body}</body>"
             return HTML(result)
 
         table = self.dataframe
 
-        self.rendered_without_reasons = _render_html(table)
+        self.rendered_without_reasons = _render_html(table, self.cherry_pick)
```

Comment:
nit: we dont need to keep passing around the `self.cherry_pick` arg to each function, since we can just access within the function as `self.cherry_pick`

Reply:
Good point!! Will get rid of that.